fiftyone/utils/transformers.py

```diff
@@ -0,0 +1,388 @@
+        if config.model is not None:
+            return config.model
+
+        return transformers.AutoModel.from_pretrained(config.name_or_path)
+
+    def predict(self, arg):
+        raise NotImplementedError("Subclass must implement predict()")
+
+
+class FiftyOneTransformerForImageClassification(FiftyOneTransformer):
+    """FiftyOne wrapper around a ``transformers.models`` model for image
+    classification.
+
+    Args:
+        config: a `FiftyOneTransformerConfig`
+    """
+
+    def _load_model(self, config):
+        if config.model is not None:
+            return config.model
+
+        return transformers.AutoModelForImageClassification.from_pretrained(
+            config.name_or_path
+        )
+
+    def _predict(self, inputs):
+        with torch.no_grad():
+            results = self.model(**inputs)
+        return to_classification(results, self.model.config.id2label)
+
+    def predict(self, arg):
+        inputs = self.image_processor(arg, return_tensors="pt")
+        return self._predict(inputs)
+
+    def predict_all(self, args):
+        inputs = self.image_processor(args, return_tensors="pt")
+        return self._predict(inputs)
+
+
+class FiftyOneTransformerForObjectDetection(FiftyOneTransformer):
+    """FiftyOne wrapper around a ``transformers.models`` model for object
+    detection.
+
+    Args:
+        config: a `FiftyOneTransformerConfig`
+    """
+
+    def _load_model(self, config):
+        if config.model is not None:
+            return config.model
+
+        return transformers.AutoModelForObjectDetection.from_pretrained(
+            config.name_or_path
+        )
+
+    def _predict(self, inputs, target_sizes):
+        with torch.no_grad():
+            outputs = self.model(**inputs)
+
+        results = self.image_processor.post_process_object_detection(
+            outputs, target_sizes=target_sizes
+        )
+        image_shapes = [i[::-1] for i in target_sizes]
+        return to_detections(results, self.model.config.id2label, image_shapes)
+
+    def predict(self, arg):
+        target_sizes = [arg.shape[:-1][::-1]]
+        inputs = self.image_processor(arg, return_tensors="pt")
+        return self._predict(inputs, target_sizes)
+
+    def predict_all(self, args):
+        target_sizes = [i.shape[:-1][::-1] for i in args]
+        inputs = self.image_processor(args, return_tensors="pt")
+        return self._predict(inputs, target_sizes)
+
+
+class FiftyOneTransformerForSemanticSegmentation(FiftyOneTransformer):
+    """FiftyOne wrapper around a ``transformers.models`` model for semantic
+    segmentation.
+
+    Args:
+        config: a `FiftyOneTransformerConfig`
+    """
+
+    def _load_model(self, config):
+        if config.model is not None:
+            return config.model
+
+        return transformers.AutoModelForSemanticSegmentation.from_pretrained(
+            config.name_or_path
+        )
+        self.mask_targets = config.model.id2label
```

Comment:
@jacobmarks line 327 is unreachable. Did you intend for this to happen? Or just delete it?

Reply:
I can take another look at `compute_embeddings()`, but in order to make it compatible with batching as well as single samples, and both image embeddings and patch embeddings, I could have sworn this was the best solution

Reply:
@brimoor and for `mask_targets`, I don't see a way for `apply_model()` to propagate `mask_targets` from a model to the dataset. I put this here in the hopes that we could figure something out, as I think this would be useful for other segmentation models as well, unless I'm missing something

Reply:
Line 327 doesn't get executed as currently written, it is after the `return` statement

Reply:
Having `model.mask_targets` is totally fine, here's an existing example of that:

```py
model = foz.load_zoo_model("deeplabv3-resnet101-coco-torch")
dataset.default_mask_targets = model.mask_targets
```

Reply:
ahhh I see. Thank you!